### PR TITLE
Full Mesh Traffic Sanity verification

### PR DESCRIPTION
**Pull request by sreejithsreekumaran**
### Description of PR

Summary:
Fixes # (issue)

A test to check reachability from any source port to any destination port in the device for each traffic class.

### Type of change

New test case to verify traffic sanity

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311

#### How did you do it?

ptf64 topology is used to have 64 ports of the DUT connected
to 64 PTF ports.

#### How did you verify/test it?

Ran the test on a T1 ptf-64 topology

#### Any platform specific information?

This is a Cisco specific test case

### CODE REVIEW

**Comment by alpeshspatel** [comment]
@kevinwangsk FYI

**Comment by kevinskwang** [comment]
@sreejithsreekumaran pls upload the test results  based on your branch full_mesh_traffic_sanity

**Comment by sreejithsreekumaran** [comment]
> @sreejithsreekumaran pls upload the test results based on your branch full_mesh_traffic_sanity

Below is the snippet from the log file. I am attaching the run logs as well as the ptf logs

[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details
collected 3 items                                                                                                                                                     

qos/test_qos_sai.py::TestQosSai::testQosSaiFullMeshTrafficSanity[single_asic-None] PASSED                                                                       [ 33%]
qos/test_qos_sai.py::TestQosSai::testQosSaiFullMeshTrafficSanity[single_dut_multi_asic-None] SKIPPED (Did not find any frontend node that is multi-asic - s...) [ 66%]
qos/test_qos_sai.py::TestQosSai::testQosSaiFullMeshTrafficSanity[multi_dut-None] SKIPPED (Don't have 2 frontend nodes - so can't run multi_dut tests)           [100%]

========================================================================== warnings summary ===========================================================================
../../usr/local/lib/python3.8/dist-packages/_yaml/__init__.py:18
  /usr/local/lib/python3.8/dist-packages/_yaml/__init__.py:18: DeprecationWarning: The _yaml extension module is now located at yaml._yaml and its location is subject to change.  To use the LibYAML-based parser and emitter, import from `yaml`: `from yaml import CLoader as Loader, CDumper as Dumper`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------- generated xml file: /run_logs/sree/tr_2024-03-21-11-21-09.xml ----------------------------------------------------
======================================================================= short test summary info =======================================================================
SKIPPED [1] qos/qos_sai_base.py:613: Did not find any frontend node that is multi-asic - so can't run single_dut_multi_asic tests
SKIPPED [1] qos/qos_sai_base.py:623: Don't have 2 frontend nodes - so can't run multi_dut tests
======================================================== 1 passed, 2 skipped, 1 warning in 4578.74s (1:16:18) =========================================================
INFO:root:Can not get Allure report URL. Please check logs

[full_mesh_ptf_logs.tar.gz](https://github.com/sonic-net/sonic-mgmt/files/14695424/full_mesh_ptf_logs.tar.gz)
[fullmesh_run_log.tar.gz](https://github.com/sonic-net/sonic-mgmt/files/14695425/fullmesh_run_log.tar.gz)

**Review by rbpittman — APPROVED** [review]

**Comment by sreejithsreekumaran** [comment]
@kevinskwang please suggest the next step. Further, can you please resolve the conflicts?

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -1098,6 +1098,12 @@ qos/test_qos_sai.py::TestQosSai::testQosSaiDwrrWeightChange:
     conditions:
       - "asic_type in ['mellanox']"
 
+qos/test_qos_sai.py::TestQosSai::testQosSaiFullMeshTrafficSanity:
+  skip:
+    reason: "Unsupported testbed type."
+    conditions:
+      - "topo_name not in ['ptf64'] and asic_type in ['cisco-8000']"
```
Comment (kevinskwang) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1105]: From your code here, the case should be skipped if asic type is cisco-8000, how you get the test result you posted?

Comment (sreejithsreekumaran) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1105]: @kevinskwang the case will be skipped only if it's not a ptf64 topology and asic-type is cisco-8000.  The test result is posted from a testbed having ptf64 topology and asic_type is cisco-8000

Comment (sreejithsreekumaran) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1145]: yes it can be used by other platforms as long as it is from Cisco.

Comment (kevinskwang) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:1145]: No, I mean other platform, like broadcom. mellanox

Comment (sreejithsreekumaran) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml]: ok. I got it now. Yes, it's written currently for Cisco platform only. 

I have fixed the condition to skip for other platforms too.

**Comment by sreejithsreekumaran** [comment]
> The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request. The pre-commit check is a mandatory check, please fix detected issues.
> 
> Detailed pre-commit check results: trim trailing whitespace.................................................Passedfix end of files.........................................................Passedcheck yaml...............................................................Passedcheck for added large files..............................................Passedcheck python ast.........................................................Passedflake8...................................................................Failed- hook id: flake8- exit code: 1tests/saitests/py3/sai_qos_tests.py:1647:19: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1658:27: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1662:23: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1686:19: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1697:27: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1701:23: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1721:19: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1732:27: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1736:23: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1755:19: E275 missing whitespace after keywordtests/saitests/py3/sai_qos_tests.py:1762:35: E275 missing whitespace after keyword...[truncated extra lines, please run pre-commit locally to view full check results]
> 
> To run the pre-commit checks locally, you can follow below steps:
> 
> 1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
>    the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
>    docker container.
> 2. Ensure that the `pre-commit` package is installed:
> 
> ```
> sudo pip install pre-commit
> ```
> 
> 3. Go to repository root folder
> 4. Install the pre-commit hooks:
> 
> ```
> pre-commit install
> ```
> 
> 5. Use pre-commit to check staged file:
> 
> ```
> pre-commit
> ```
> 
> 6. Alternatively, you can check committed files using:
> 
> ```
> pre-commit run --from-ref <commit_id> --to-ref <commit_id>
> ```

@kevinskwang these pre-commit warnings are from lines which are not part of this PR #9896. Looks like the rules have changed from then to now.

**Comment by kevinskwang** [comment]
could you resolve the conflicts?

**Review by kevinskwang — DISMISSED** [review]

**Comment by sreejithsreekumaran** [comment]
> could you resolve the conflicts?

@kevinskwang  I have resolved the conflict. Please re-approve and do the needful.

**Review by kevinskwang — APPROVED** [review]

**Comment by sreejithsreekumaran** [comment]
> ptf64 topology is used to have 64 ports of the DUT connected
> to 64 PTF ports.

@kevinskwang raised PR #14678

**Pull request merged**